docs/api-diffs/9.0/rc.1/diff.md

```diff
@@ -0,0 +1,1290 @@
+* [Aspire.Hosting.Kafka](#aspirehostingkafka)
+* [Aspire.Hosting.Milvus](#aspirehostingmilvus)
+* [Aspire.Hosting.MongoDB](#aspirehostingmongodb)
+* [Aspire.Hosting.MySql](#aspirehostingmysql)
+* [Aspire.Hosting.Nats](#aspirehostingnats)
+* [Aspire.Hosting.NodeJs](#aspirehostingnodejs)
+* [Aspire.Hosting.Oracle](#aspirehostingoracle)
+* [Aspire.Hosting.PostgreSQL](#aspirehostingpostgresql)
+* [Aspire.Hosting.Python](#aspirehostingpython)
+* [Aspire.Hosting.RabbitMQ](#aspirehostingrabbitmq)
+* [Aspire.Hosting.Redis](#aspirehostingredis)
+* [Aspire.Hosting.SqlServer](#aspirehostingsqlserver)
+* [Aspire.OpenAI](#aspireopenai)
+
+## Aspire.Azure.AI.OpenAI
+
+``` diff
+ {
+     namespace Aspire.Azure.AI.OpenAI {
+         public sealed class AzureOpenAISettings : IConnectionStringSettings {
++            public bool DisableMetrics { get; set; }
+         }
+     }
+     namespace Microsoft.Extensions.Hosting {
++        public static class AspireConfigurableOpenAIExtensions {
++            public static void AddKeyedOpenAIClientFromConfiguration(this IHostApplicationBuilder builder, string name);
++            public static void AddOpenAIClientFromConfiguration(this IHostApplicationBuilder builder, string connectionName);
++        }
+     }
+ }
+```
+
+## Aspire.Hosting
+
+``` diff
+ {
+     namespace Aspire.Hosting {
+         public static class ContainerResourceBuilderExtensions {
+-            public static IResourceBuilder<ContainerResource> AddContainer(this IDistributedApplicationBuilder builder, string name, string image);
++            public static IResourceBuilder<ContainerResource> AddContainer(this IDistributedApplicationBuilder builder, [ResourceNameAttribute] string name, string image);
+-            public static IResourceBuilder<ContainerResource> AddContainer(this IDistributedApplicationBuilder builder, string name, string image, string tag);
++            public static IResourceBuilder<ContainerResource> AddContainer(this IDistributedApplicationBuilder builder, [ResourceNameAttribute] string name, string image, string tag);
+-            public static IResourceBuilder<ContainerResource> AddDockerfile(this IDistributedApplicationBuilder builder, string name, string contextPath, string? dockerfilePath = null, string? stage = null);
++            public static IResourceBuilder<ContainerResource> AddDockerfile(this IDistributedApplicationBuilder builder, [ResourceNameAttribute] string name, string contextPath, string? dockerfilePath = null, string? stage = null);
++            public static IResourceBuilder<T> WithContainerName<T>(this IResourceBuilder<T> builder, string name) where T : ContainerResource;
++            public static IResourceBuilder<T> WithLifetime<T>(this IResourceBuilder<T> builder, ContainerLifetime lifetime) where T : ContainerResource;
+         }
+         public class DistributedApplicationBuilder : IDistributedApplicationBuilder {
++            public string AppHostPath { get; }
```

Comment:
@davidfowl @danegsta - Was this new public property necessary? I see we didn't add it on the interface, was that intentional?

Reply:
Honestly, looking at the original PR again, this property is probably superfluous, since we ended up generating the project hash in the DistributedApplicationBuilder constructor and exposing it as an in memory configuration property.

Reply:
Let's remove it.